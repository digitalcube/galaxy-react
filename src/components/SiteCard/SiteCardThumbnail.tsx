import React, { FC } from 'react';
import { Screenshot } from '../Screenshot/Screenshot';

export type SiteCardThumbnailProps = React.DetailedHTMLProps<React.ImgHTMLAttributes<HTMLImageElement>, HTMLImageElement>

export const SiteCardThumbnail: FC<SiteCardThumbnailProps> = ({src, ...imageProps}) => {
  if (!src) return null;
  return (
    <div className="site-thumbnail rounded-left">
      {/(png|jpg|jpeg|svg)$/.test(src) ? (
      <img src={src} {...imageProps} width="167" height="111"  />
      ): (
      <Screenshot url={src} {...imageProps} width="167" height="111"  />
      )}
    </div>
  )
}
