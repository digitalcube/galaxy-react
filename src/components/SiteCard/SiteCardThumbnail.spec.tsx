import { render } from '@testing-library/react';
import React from 'react';
import { Default as SiteCardThumbnail, } from './SiteCardThumbnail.stories';

describe('SiteCardThumbnail', () => {
  it('to match snapshot [column %p | item %p]', () => {
    const { asFragment } = render(
      <SiteCardThumbnail {...(SiteCardThumbnail.args as any)} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
