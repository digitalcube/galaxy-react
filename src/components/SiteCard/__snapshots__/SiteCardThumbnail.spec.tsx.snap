// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteCardThumbnail to match snapshot [column %p | item %p] 1`] = `
<DocumentFragment>
  <div
    class="site-thumbnail rounded-left"
  >
    <img
      alt="site thumbnail"
      height="111"
      src="/images/site-a.png"
      width="167"
    />
  </div>
</DocumentFragment>
`;
